test/mixed.js

```diff
@@ -483,6 +483,64 @@ describe('Mixed Types ', () => {
     called.should.equal(true);
   });
 
+  it('tests should be able to access nested parent', async () => {
+    let calledFirst = false;
+    let calledThird = false;
+
+    let testFixture = {
+      firstField: 'test',
+      second: [
+        {
+          thirdField: 'test3',
+        },
+        {
+          thirdField: 'test4',
+        },
+      ],
+    };
+
+    let third = object({
+      thirdField: mixed().test({
+        test() {
+          calledThird = true;
+
+          // I want to do something like this,
+          // this.resolve(ref(this.path),this.from[1].value);
+          // OR probably preferably
+          // this.from[1].resolve(ref(this.path))
+          // but I can't get it working :( ...help?
+
+          // so I added an "index" option to array test options to make this test make sense
+          this.from[0].value.should.eql(testFixture.second[this.options.index]);
+          this.from[0].schema.should.equal(third);
+
+          this.from[1].value.should.equal(testFixture);
+          this.from[1].schema.should.equal(first);
+
+          return true;
+        },
+      }),
+    });
+
+    let second = array().of(third);
+
+    let first = object({
+      firstField: mixed().test({
+        test() {
+          calledFirst = true;
+          this.from[0].value.should.equal(testFixture);
+          this.from[0].schema.should.equal(first);
+          return true;
+        },
+      }),
+      second,
+    });
+
+    await first.validate(testFixture);
+    calledFirst.should.equal(true);
+    calledThird.should.equal(true);
```

Comment:
I'm still find this test hard to understand. Maybe b/c it's large. If the goal is to assert that the `from` value is correct maybe instead of all the flags, have the deeply nested schema save off the `from` value via a closure then assert after the validate here that it matches what you expect?

Reply:
Done!